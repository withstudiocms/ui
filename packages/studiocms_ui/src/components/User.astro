---
import { Image } from 'astro:assets';
import Icon from '../utils/Icon.astro';

interface Props {
	name: string;
	description: string;
	avatar?: string;
	class?: string;
  loading?: 'eager' | 'lazy'; 
};

const { name, description, avatar, class: className, loading } = Astro.props;
---
<div class="sui-user-container" class:list={[ className ]}>
  <div class="sui-avatar-container">
    {avatar ? (
      <Image src={avatar} inferSize loading={loading} alt={name} class="sui-avatar-img" />
    ) : (
      <Icon name='user' width={24} height={24} />
    )}
  </div>
  <div class="sui-text-content">
    <span class="sui-name">{name}</span>
    <span class="sui-description">{description}</span>
  </div>
</div>
<style>
  .sui-user-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: 1rem;
  }

  .sui-avatar-container {
    width: 2.5rem;
    height: 2.5rem;
    background-color: hsl(var(--default-base));
    border-radius: 3rem;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;
    border: 1px solid hsl(var(--border));
  }

  .sui-avatar-img {
    width: 100%;
    height: auto;
  }

  .sui-text-content {
    display: flex;
    flex-direction: column;
    gap: .125rem;
  }

  .sui-name {
    font-size: 1em;
    font-weight: 600;
  }

  .sui-description {
    font-size: .875em;
    font-weight: 400;
    color: hsl(var(--text-muted));
  }
</style>
