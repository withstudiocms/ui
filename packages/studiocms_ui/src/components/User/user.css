.sui-user-container {
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 1rem;
}

.sui-avatar-container {
  width: 2.5rem;
  height: 2.5rem;
  background-color: hsl(var(--default-base));
  border-radius: var(--radius-full);
  display: flex;
  align-items: center;
  justify-content: center;
  overflow: hidden;
  border: 1px solid hsl(var(--border));
}

.sui-avatar-img {
  width: 100%;
  height: auto;
}

.sui-text-content {
  display: flex;
  flex-direction: column;
  gap: .125rem;
}

.sui-name {
  font-size: 1em;
  font-weight: 600;
}

.sui-description {
  font-size: .875em;
  font-weight: 400;
  color: hsl(var(--text-muted));
}
