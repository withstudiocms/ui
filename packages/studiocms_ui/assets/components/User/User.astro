---
// @ts-ignore
import { Image } from 'astro:assets';
import Icon from '../Icon/Icon.astro';
import './user.css';

/**
 * The props for the User component.
 */
interface Props {
	/**
	 * The name of the user.
	 */
	name: string;
	/**
	 * The description of the user. Could be a role, a handle, etc.
	 */
	description: string;
	/**
	 * The avatar of the user. Either a URL to an image or an imported image.
	 */
	avatar?: string;
	/**
	 * Additional classes to apply to the user container.
	 */
	class?: string;
	/**
	 * The loading strategy for the image. Defaults to `lazy`.
	 */
	loading?: 'eager' | 'lazy';
}

const { name, description, avatar, class: className, loading = 'lazy' } = Astro.props;
---
<div class="sui-user-container" class:list={[ className ]}>
  <div class="sui-avatar-container">
    {avatar ? (
      <Image src={avatar} inferSize loading={loading} alt={name} class="sui-avatar-img" />
    ) : (
      <Icon name='user' width={24} height={24} role='img' aria-label={`Placeholder avatar for ${name}`} />
    )}
  </div>
  <div class="sui-text-content">
    <span class="sui-name">{name}</span>
    <span class="sui-description">{description}</span>
  </div>
</div>
