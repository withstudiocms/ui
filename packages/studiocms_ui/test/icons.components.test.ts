import { describe, expect } from 'vitest';
import Checkmark from '../src/icons/Checkmark.astro';
import ChevronUpDown from '../src/icons/ChevronUpDown.astro';
import User from '../src/icons/User.astro';
import XMark from '../src/icons/X-Mark.astro';
import { test } from './fixtures/vitest/AstroContainer';

describe('Icon Components', () => {
	test('Checkmark component renders correctly', async ({ renderComponent }) => {
		const result = await renderComponent(Checkmark, 'Checkmark');
		expect(result).toMatchSnapshot();
	});

	test('ChevronUpDown component renders correctly', async ({ renderComponent }) => {
		const result = await renderComponent(ChevronUpDown, 'ChevronUpDown');
		expect(result).toMatchSnapshot();
	});

	test('User component renders correctly', async ({ renderComponent }) => {
		const result = await renderComponent(User, 'User');
		expect(result).toMatchSnapshot();
	});

	test('XMark component renders correctly', async ({ renderComponent }) => {
		const result = await renderComponent(XMark, 'XMark');
		expect(result).toMatchSnapshot();
	});
});
