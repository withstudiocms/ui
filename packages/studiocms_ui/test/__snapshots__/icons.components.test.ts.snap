// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon Components > Checkmark component renders correctly 1`] = `"<svg role="presentation" viewBox="0 0 17 18" width="24" height="24"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg>"`;

exports[`Icon Components > ChevronUpDown component renders correctly 1`] = `"<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" width="24" height="24" stroke-width="1.5" stroke="currentColor"> <path stroke-linecap="round" stroke-linejoin="round" d="M8.25 15 12 18.75 15.75 15m-7.5-6L12 5.25 15.75 9"></path> </svg>"`;

exports[`Icon Components > User component renders correctly 1`] = `"<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" width="24" height="24"> <path stroke-linecap="round" stroke-linejoin="round" d="M15.75 6a3.75 3.75 0 1 1-7.5 0 3.75 3.75 0 0 1 7.5 0ZM4.501 20.118a7.5 7.5 0 0 1 14.998 0A17.933 17.933 0 0 1 12 21.75c-2.676 0-5.216-.584-7.499-1.632Z"></path> </svg>"`;

exports[`Icon Components > XMark component renders correctly 1`] = `"<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" width="24" height="24"> <path stroke-linecap="round" stroke-linejoin="round" d="M6 18 18 6M6 6l12 12"></path> </svg>"`;
